pkg/nat/resolver.go

```diff
@@ -0,0 +1,68 @@
+package nat
+
+import (
+	"context"
+	"fmt"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/pkg/errors"
+
+	"github.com/giantswarm/capg-firewall-rule-operator/pkg/google"
+	"github.com/giantswarm/capg-firewall-rule-operator/pkg/k8sclient"
+)
+
+func NewIPResolver(gcpClusters *k8sclient.GCPCluster, addresses *compute.AddressesClient, routers *compute.RoutersClient) *IPResolver {
+	return &IPResolver{
+		gcpClusters: gcpClusters,
+		addresses:   addresses,
+		routers:     routers,
+	}
+}
+
+type IPResolver struct {
+	gcpClusters *k8sclient.GCPCluster
+	addresses   *compute.AddressesClient
+	routers     *compute.RoutersClient
+}
+
+func (r *IPResolver) GetIPs(ctx context.Context, managementCluster types.NamespacedName) ([]string, error) {
+	cluster, err := r.gcpClusters.Get(ctx, managementCluster)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	if google.IsNilOrEmpty(cluster.Status.Network.Router) {
+		return nil, fmt.Errorf("cluster %s/%s does not have router yet", managementCluster.Namespace, managementCluster.Name)
```

Comment:
should we requeue in this case? I don't think this will even happen, but when the management cluster gets updated with is router, the workload cluster that we are reconciling won't be automatically reconciled.

Reply:
Right, that needs to be checked [here](https://github.com/giantswarm/capg-firewall-rule-operator/blob/202323b2d3c875ac5bb76b1d3facf815c81b35df/controllers/gcpcluster.go#L103-L111). That block of code is leaking some abstraction though.. I can't think of a good way to fix it at the moment though. Maybe the two reconcilers can have a function like `CheckResourcesExist(cluster *capg.GCPCluster)` or something, that's called before adding the finalizer.

In any case though I think this should stay in the `nat.IPResolver`. I view it as isolated from the rest of the code (because it needs to be integration tested) and if the router isn't there it will panic.

So WDYT? Just adding the check in the controller for now?

Reply:
Adding the check in the controller sounds fine.

Reply:
Okay I'm an idiot.. this isn't the WC's router (the one we're reconciling). This is the MC's router, so we have no choice but to error here. :facepalm: 

I'll revert the last commit